src/crdt.rs

```diff
@@ -1353,6 +1354,55 @@ impl TestNode {
             },
         }
     }
+    pub fn new_with_external_ip(pubkey: Pubkey, ip: IpAddr) -> TestNode {
+        fn bind() -> (u16, UdpSocket) {
+            match udp_random_bind(8000, 10000, 5) {
```

Comment:
Should we use 8001, or maybe 8100, instead of 8000?  8000 is hard coded in places like https://github.com/solana-labs/solana/blob/bca22946556bc0243f035cf4e8eaf1531ee36745/src/bin/fullnode.rs#L55

Reply:
I have changed it to 8100 for now. I think, going forward, only NCP port for leader needs to be hardcoded. If someday we add a DNS entry with SRV record, we can potentially remove that as well.